src/jormungandr/jormungandr/src/fragment/pool.rs

```diff
@@ -553,95 +474,42 @@ pub(super) mod internal {
             &mut self,
             fragments: impl IntoIterator<Item = (Fragment, FragmentId)>,
         ) -> Vec<(Fragment, FragmentId)> {
-            let max_fragments = self.max_entries - self.entries.len();
             fragments
                 .into_iter()
                 .filter(|(fragment, id)| {
                     if self.entries.contains(id) {
                         false
                     } else {
+                        if self.entries.len() >= self.max_entries && self.entries.len() != 0 {
+                            // Remove an oldest entry from the pool
+                            let (_, fragment) = self.entries.pop_back().expect("entry must exist");
```

Comment:
The fragments are being inserted at the front so, to make it FIFO, this should use `pop_front` instead. Did I get this right?

Reply:
Saw it at line 489.

Reply:
Oh right, makes sense. My bad. Thanks for clarifying